protocol_rest.go

```diff
@@ -441,6 +443,15 @@ func restIsHTTPBody(msg protoreflect.MessageDescriptor, bodyPath []protoreflect.
 	return msg != nil && msg.FullName() == "google.api.HttpBody"
 }
 
+// restFieldGetter returns a field descriptor by name or JSON name.
+func restFieldGetter(fds protoreflect.FieldDescriptors, name protoreflect.Name) protoreflect.FieldDescriptor {
+	fd := fds.ByName(name)
+	if fd == nil {
+		fd = fds.ByJSONName(string(name))
+	}
```

Comment:
For consistency with `protojson`, I would swap these: try the JSON name first, and then fallback to original name if nothing found.

Reply:
Should we encode the query params by default as JSON too? I do think that would be more consistent with the body.

Reply:
We now prefer JSON names on decoding query parameters, and encode query parameters using their JSON form.

Reply:
Yeah, good call.